features/lwipstack/LWIPStack.cpp

```diff
@@ -439,7 +439,7 @@ nsapi_size_or_error_t LWIP::socket_sendto(nsapi_socket_t handle, const SocketAdd
     }
     if (netif_) {
         if ((addr.version == NSAPI_IPv4 && !get_ipv4_addr(netif_)) ||
-                (addr.version == NSAPI_IPv6 && !get_ipv6_addr(netif_))) {
+                (addr.version == NSAPI_IPv6 && !get_ipv6_addr(netif_) && !get_ipv6_link_local_addr(netif_))) {
```

Comment:
I'm sure this "let's try to precheck before sending" test was added with good intentions, but it keeps causing grief like this. Why are we trying to second-guess what the IP stack will accept?

Reply:
Do you suggest to remove this complete check?
if ((addr.version == NSAPI_IPv4 && !get_ipv4_addr(netif_)) ||
(addr.version == NSAPI_IPv6 && !get_ipv6_addr(netif_) && !get_ipv6_link_local_addr(netif_))) {

Reply:
It's my gut feeling that we shouldn't be getting involved with this sort of test,  but I'm not familiar with the details of what the test was trying to solve.

I think it may have been an attempt to make sure an error was generated with an address type mismatch to pass some socket test, because lwIP just silently drops the packet itself?

I think we should leave it for now - I was just having a general grumble at the concept. If it is what I think it is, then we'd need to make an lwIP internal fix to remove it.

Reply:
Ok will keep the change as it is for now.